docs/tutorials/web/ep-webnn.md

```diff
@@ -74,59 +74,59 @@ To use WebNN EP, you just need to make 3 small changes:
-If you get your input data from a MLBuffer, or you want to keep the output data on MLBuffer for further processing, you can use IO binding to keep the data on MLBuffer. This will be especially helpful when running transformer based models, which usually runs a single model multiple times with previous output as the next input.
+By default, a model's inputs and outputs are tensors that hold data in CPU memory. When you run a session with WebNN EP with 'gpu' or 'npu' device type, the data is copied to GPU or NPU memory, and the results are copied back to CPU memory. Memory copy between different devices as well as different sessions will bring much overhead to the inference time, WebNN provides a new opaque device-specific storage type MLTensor to address this issue.
+If you get your input data from a MLTensor, or you want to keep the output data on MLTensor for further processing, you can use IO binding to keep the data on MLTensor. This will be especially helpful when running transformer based models, which usually runs a single model multiple times with previous output as the next input.
 
-For model input, if your input data is a WebNN storage MLBuffer, you can [create a MLBuffer tensor and use it as input tensor](#create-input-tensor-from-a-mlbuffer).
+For model input, if your input data is a WebNN storage MLTensor, you can [create a MLTensor tensor and use it as input tensor](#create-input-tensor-from-a-mltensor).
 
 For model output, there are 2 ways to use the IO binding feature:
-- [Use pre-allocated MLBuffer tensors](#use-pre-allocated-mlbuffer-tensors)
+- [Use pre-allocated MLTensor tensors](#use-pre-allocated-mltensor-tensors)
 - [Specify the output data location](#specify-the-output-data-location)
 
 Please also check the following topic:
-- [MLBuffer tensor life cycle management](#mlbuffer-tensor-life-cycle-management)
+- [MLTensor tensor life cycle management](#mltensor-tensor-life-cycle-management)
 
-**Note:** The MLBuffer necessitates a shared MLContext for IO binding. This implies that the MLContext should be pre-created as a WebNN EP option and utilized across all sessions.
+**Note:** The MLTensor necessitates a shared MLContext for IO binding. This implies that the MLContext should be pre-created as a WebNN EP option and utilized across all sessions.
 
-### Create input tensor from a MLBuffer
+### Create input tensor from a MLTensor
 
-If your input data is a WebNN storage MLBuffer, you can create a MLBuffer tensor and use it as input tensor:
+If your input data is a WebNN storage MLTensor, you can create a MLTensor tensor and use it as input tensor:
 
 ```js
 const mlContext = await navigator.ml.createContext({deviceType, ...});
-const inputMLBuffer = await mlContext.createBuffer({
+const inputMLTensor = await mlContext.createTensor({
   dataType: 'float32',
   dimensions: [1, 3, 224, 224],
-  usage: MLBufferUsage.WRITE_TO,
+  usage: MLTensorUsage.WRITE_TO,
```

Comment:
FYI, I think this was renamed to `MLTensorUsage.WRITE` (and `READ_FROM` to `READ`) per https://github.com/webmachinelearning/webnn/pull/754/files, but if we just want to keep it to the rename, that's fine.

Reply:
Oh, but the implementation hasn't changed, I will follow up in a new PR.